### PR TITLE
refactoring fn_typeOfSoldier.sqf

### PR DESCRIPTION
**Pull request by AlexTriada**
### What have you changed and why?
Information: 
I remade this file, left the old logic (refactoring),
to more controlling and understanding what is going on,
for my work with fixing bugs - maybe someone will be interested

tested, working fine, without errors

### CODE REVIEW

File: A3-Antistasi/functions/AI/fn_typeOfSoldier.sqf [hunk 1]
```diff
@@ -1,106 +1,115 @@
...
+
+    private _haveMines = (magazines _unit) findIf
+        { (_x call BIS_fnc_itemType) select 0 == "Mine" } != -1;
+
+    case (_haveMines): { "Engineer" };
+
+    private _isAA = false;
+    private _isAT = secondaryWeapon _unit != ""
+        && { private _sWBase = [secondaryWeapon _unit] call BIS_fnc_baseWeapon;
+            private _magTypeAT = (getArray SW_BASE_MAGS) # 0;
```
Comment (HakonRydland) [inline A3-Antistasi/functions/AI/fn_typeOfSoldier.sqf:65]: macro should pass in type as argument, example:
```sqf
#define wepMags(_weapon) getArray (configFile >> "CfgWeapons" >> _weapon >> "magazines")
```
```sqf
private _launcherBase = [secondaryWeapon _unit] call BIS_fnc_baseWeapon;
private _magTypeAT = wepMags(_launcherBase)#0;
```
but is functionally fine

File: A3-Antistasi/functions/AI/fn_typeOfSoldier.sqf [hunk 2]
```diff
@@ -1,106 +1,115 @@
...
+            (magazinesAmmoFull _unit) findIf { _x # 0 == _magTypeAT } != -1
+        && { _isAA = true;
```
Comment (HakonRydland) [inline A3-Antistasi/functions/AI/fn_typeOfSoldier.sqf:67]: original logic is maintained, but its honestly a mess, works fine though and as its a refactor im ok leaving it as is

**Review by HakonRydland — APPROVED** [review]
functionality is fine, a few things that could be improved, macros should be passing the type as argument imo, even though its only one specific type were working with currently this may change and its more readable as given in the example. a lot of the more complex type decisions are not ideal with how its written but definitely a mayor improvement to the original.

too summarize, functionality is maintained, readability as improved considerably, with only a bit more room for improvement imo.

if you want to improve upon it based on my comments i leave completely up to you.

**Comment by Bob-Murphy** [comment]
@AlexTriada please adapt based on the comments.
Thank you.

**Review by HakonRydland — APPROVED** [review]

**Pull request merged**